contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.3.0--4.4.0.sql

```diff
@@ -10566,6 +10566,50 @@ CALL sys.babelfish_alter_default_privilege_on_schema();
 -- Drop this procedure after it gets executed once.
 DROP PROCEDURE sys.babelfish_alter_default_privilege_on_schema();
 
+-- To handle the typmod values for procedure and function
+UPDATE pg_proc p1
+SET probin = 
+    CASE 
+        WHEN p1.probin::jsonb->'typmod_array' = '[]'::jsonb THEN p1.probin
```

Comment:
Instead of this lets add filter in where clause using proargtypes and prorettype.

Reply:
Updating the filter with the following two changes and resoning:

1. In procedure we only rely on proargtypes, if its not there then typmod_array will be [] , so exclude this in where clause.

2. In function we will always have a return type thus there will always be some value in typmod_array. In function when its ITVF or MSTVF, then we have values in proallargtypes and not in proargtypes , but we dont need to update these kinds of functions therefore we can exclude this as well in where clause